### PR TITLE
Nested E2E: Add filters for broker-specific tests

### PR DESCRIPTION
**Pull request by and-rewsmith**
Three things in this PR:
1. Our nested E2E tests are running broker-specific tests without the broker which is causing test failures (i.e. BridgeConfigUpdate). We need to implement a filter so that we don't run these broker specific tests without the broker enabled. I named this filter ```BrokerRequired```.
2. I am also adding a filter to the PlugAndPlay (```LegacyMqttRequired```)tests to only run with the legacy mqtt protocol head. These currently have very confusing logic around an extra broker enabled flag that always gets ignored. A filter will work much better.
3. Some tests were manually being filtered because of an outstanding PBI. The PBI is now done so I have removed the manual filter.

I think in general we should move to replace these manual test filters with Category filters. I am not sure why are currently filtering this:
```
FullyQualifiedName!~PriorityQueueModuleToHubMessages&FullyQualifiedName!~SasOutOfScope&FullyQualifiedName!~X509ManualProvision&FullyQualifiedName!~AuthorizationPolicyUpdateTest&FullyQualifiedName!~AuthorizationPolicyExplicitPolicyTest'
```

### CODE REVIEW

File: builds/e2e/templates/e2e-run.yaml [hunk 1]
```diff
@@ -24,14 +24,15 @@ steps:
     {
       $filter += '&Category!=SingleNodeOnly'
       $filter += '&Category!=NestedEdgeAmqpOnly'
+      $filter += '&Category!=LegacyMqttRequired'
     
-      #We are disabling some additional tests. See  PBI #9171870. Once this is fixed we should run what's directly below.
-      #$filter += '&FullyQualifiedName!~Provisioning&FullyQualifiedName!~PlugAndPlay&FullyQualifiedName!~PriorityQueueModuleToHubMessages&FullyQualifiedName!~SasOutOfScope'
       $filter += '&FullyQualifiedName!~Provisioning&FullyQualifiedName!~PlugAndPlay&FullyQualifiedName!~PriorityQueueModuleToHubMessages&FullyQualifiedName!~SasOutOfScope&FullyQualifiedName!~X509ManualProvision&FullyQualifiedName!~AuthorizationPolicyUpdateTest&FullyQualifiedName!~AuthorizationPolicyExplicitPolicyTest'
```
Comment (and-rewsmith) [inline builds/e2e/templates/e2e-run.yaml:29]: I should remove PlugAndPlay from the manual filter as it now has a named filter

**Comment by and-rewsmith** [comment]
Regular E2E is passing everything except validate metrics, which is known to be flaky:
https://msazure.visualstudio.com/One/_build/results?buildId=41699162&view=ms.vss-test-web.build-test-results-tab

Nested E2E is failing some tests, but these failures are not related to this change:
https://msazure.visualstudio.com/One/_build/results?buildId=41699162&view=ms.vss-test-web.build-test-results-tab

File: builds/e2e/templates/e2e-run.yaml (hunk 1 above)
Comment (and-rewsmith) [inline builds/e2e/templates/e2e-run.yaml:29]: @vadim-kovalyov I chose to remove only the line directly below when we realized this PBI was completed. Therefore the AuthorizationPolicyExplicitTest is still filtered.

I tried to run the tests with both filter lines removed and the tests hung due to some Provisioning test.

Comment (vadim-kovalyov) [reply builds/e2e/templates/e2e-run.yaml]: hm, I still wonder why AuthZ tests are filtered out. If Broker is enabled they should work in tested. I has been manually verified during bug bash. It is basically just a twin update...

**Review by vadim-kovalyov — APPROVED** [review]
This is great @and-rewsmith, thanks for doing that. Categories will work much better, and I'm glad you figured out that Bridge tests are failing b/c of the broker. When I added that I dit not realize that nested tests do not have broker enabled...

**Comment by and-rewsmith** [comment]
@vadim-kovalyov It is confusing from the yaml, but for nested we first run mqtt nested then run amqp nested. The bridge tests fail for the second e2e run, but I left them enabled for the first mqtt run (and they pass).

I think further work is needed to determine:
- Why are we filtering this?
FullyQualifiedName!~PriorityQueueModuleToHubMessages&FullyQualifiedName!~SasOutOfScope&FullyQualifiedName!~X509ManualProvision&FullyQualifiedName!~AuthorizationPolicyUpdateTest&FullyQualifiedName!~AuthorizationPolicyExplicitPolicyTest
- Why do we have flaky tests?
- How can we fix the recent proxy tests being re-enabled from causing timeouts

**Comment by vadim-kovalyov** [comment]
Good questions...

>     * Why are we filtering this?
>       FullyQualifiedName!~PriorityQueueModuleToHubMessages&FullyQualifiedName!~SasOutOfScope&FullyQualifiedName!~X509ManualProvision&FullyQualifiedName!~AuthorizationPolicyUpdateTest&FullyQualifiedName!~AuthorizationPolicyExplicitPolicyTest
 
Maybe @huguesBouvier can shed some light. Looks like he's the one who added that filter.

>     * Why do we have flaky tests?

¯\_(ツ)_/¯
 
>     * How can we fix the recent proxy tests being re-enabled from causing timeouts?

I think the failure caused by changing the protocol to WS. Maybe we can try to repro it locally and see why WS does not work for PnP tests. @dylanbronson, maybe you have ideas why PnP may not work over WS?

**Comment by and-rewsmith** [comment]
@vadim-kovalyov I already made a devops bug for the unknown filters and @ hugues on it. Should get triaged. 

I think for the flaky tests we should open a bug for each flaky test. I was going to let this change land in master then we can open bugs for each failing test.

I'll leave the PnP investigations to dylan. Have created a bug.

**Pull request merged**